pkg/component/etcd/etcd/crd.go

```diff
@@ -9,98 +9,146 @@ import (
 		return err
 	}
 
 	if len(etcdCopyBackupsTaskList.Items) > 0 {
-		return fmt.Errorf("cannot delete etcd CRDs because there are still druidv1alpha1.EtcdCopyBackupsTask resources left in the cluster")
+		return fmt.Errorf("cannot delete etcd CRDs because there are still druidcorev1alpha1.EtcdCopyBackupsTask resources left in the cluster")
 	}
 
-	if err := gardenerutils.ConfirmDeletion(ctx, c.client, &apiextensionsv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: etcdCopyBackupsTaskCRDName}}); client.IgnoreNotFound(err) != nil {
+	if err := gardenerutils.ConfirmDeletion(ctx, c.client, &apiextensionsv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: druidcorecrds.ResourceNameEtcdCopyBackupsTask}}); client.IgnoreNotFound(err) != nil {
 		return err
 	}
 
 	var fns []flow.TaskFn
 
-	for _, resource := range crdResources {
+	for _, resource := range c.crdGetter.GetAllCRDs() {
 		r := resource
 		fns = append(fns, func(ctx context.Context) error {
-			return client.IgnoreNotFound(c.applier.DeleteManifest(ctx, kubernetes.NewManifestReader([]byte(r))))
+			return client.IgnoreNotFound(c.client.Delete(ctx, r))
 		})
 	}
 
 	return flow.Parallel(fns...)(ctx)
 }
+
+// CRDGetter provides methods to get CRDs defined in etcd-druid.
+type CRDGetter interface {
+	// GetAllCRDs returns a map of CRD names to CRD objects.
+	GetAllCRDs() map[string]*apiextensionsv1.CustomResourceDefinition
+	// GetCRD returns the CRD with the given name.
+	// An error is returned if no CRD is found with the given name.
+	GetCRD(name string) (*apiextensionsv1.CustomResourceDefinition, error)
+}
+
+type crdGetter struct {
+	crdResources map[string]*apiextensionsv1.CustomResourceDefinition
+}
+
+var _ CRDGetter = (*crdGetter)(nil)
+
+// NewCRDGetter creates a new CRDGetter.
+func NewCRDGetter(k8sVersion *semver.Version) (CRDGetter, error) {
+	crdResources, err := getEtcdCRDs(k8sVersion)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get etcd-druid CRDs for Kubernetes version %s: %w", k8sVersion, err)
+	}
+	return &crdGetter{
+		crdResources: crdResources,
+	}, nil
+}
+
+func (c *crdGetter) GetAllCRDs() map[string]*apiextensionsv1.CustomResourceDefinition {
+	return c.crdResources
+}
+
+func (c *crdGetter) GetCRD(name string) (*apiextensionsv1.CustomResourceDefinition, error) {
+	crdObj, ok := c.crdResources[name]
+	if !ok {
+		return nil, fmt.Errorf("CRD %s not found", name)
+	}
+	return crdObj, nil
+}
+
+func getEtcdCRDs(k8sVersion *semver.Version) (map[string]*apiextensionsv1.CustomResourceDefinition, error) {
+	crdYAMLs, err := druidcorecrds.GetAll(k8sVersion.String())
+	if err != nil {
+		return nil, err
+	}
+	var crdResources = make(map[string]*apiextensionsv1.CustomResourceDefinition, len(crdYAMLs))
+	for crdName, crdYAML := range crdYAMLs {
+		crdObj, err := kubernetesutils.DecodeCRD(crdYAML)
+		if err != nil {
+			return nil, err
```

Comment:
fmt.Errorf

Reply:
addressed in https://github.com/gardener/gardener/pull/11545/commits/e469199967b1256227f4367254263ca7d2db08db